codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -20,23 +20,152 @@
+     *                   Larger values result in better compression at the expense of
+     *                   memory usage.  The default value is {@code 15}.
+     * @return the compressor itself for fluent method calls
+     */
+    public HttpContentCompressor withWindowBits(int windowBits) {
+        if (windowBits < 9 || windowBits > 15) {
+            throw new IllegalArgumentException("windowBits: " + windowBits + " (expected: 9-15)");
+        }
+        this.windowBits = windowBits;
+        return this;
+    }
+
+    /**
+     * Specifies a new memory level for this content compressor.
+     *
+     * @param memLevel how much memory should be allocated for the internal compression
+     *                 state.  {@code 1} uses minimum memory and {@code 9} uses maximum
+     *                 memory.  Larger values result in better and faster compression
+     *                 at the expense of memory usage.  The default value is {@code 8}
+     * @return the compressor itself for fluent method calls
+     */
+    public HttpContentCompressor withMemLevel(int memLevel) {
+        if (memLevel < 1 || memLevel > 9) {
+            throw new IllegalArgumentException("memLevel: " + memLevel + " (expected: 1-9)");
+        }
+        this.memLevel = memLevel;
+        return this;
+    }
+
+    /**
+     * Specifies the minimal content length for a response to be compressed.
+     *
+     * @param minCompressableContentLength the minimal content length in bytes for a response to be compressed at all.
+     *                                     Note that response without content length will always be compressed.
+     *                                     The default value is {@code 1024}
+     * @return the compressor itself for fluent method calls
+     */
+    public HttpContentCompressor withMinCompressableContentLength(int minCompressableContentLength) {
+        this.minCompressableContentLength = minCompressableContentLength;
+        return this;
+    }
+
+    /**
+     * Specifies a pattern which matches all content types to be compressed.
+     *
+     * @param compressableContentTypes a regular expression matching all content types eligible for compression. If
+     *                                 a response does not contain a content type will always be compressor. If the
+     *                                 parameter is <tt>null</tt>, the content compressor is put into legacy mode,
+     *                                 which will compress all responses independent of their length, content type
+     *                                 and response type. By default all <tt>text/*</tt> and <tt>application/json</tt>
+     *                                 are compressed.
+     * @return the compressor itself for fluent method calls
+     */
+    public HttpContentCompressor withCompressableContentTypes(Pattern compressableContentTypes) {
+        this.compressableContentTypes = compressableContentTypes;
```

Comment:
I could add another setter which accepts a plain string (and internally convert this into a regexp). The only other alternative would be to accept some kind of functional interface (String -> Boolean) but wouldn't that be kind of an overkill?

Or do you have a hint or idea how to implement that gracefully?